src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -129,21 +134,11 @@ void LambdaFormInvokers::regenerate_holder_classes() {
     char *class_name = java_lang_String::as_utf8_string(h_name());
     int len = h_bytes->length();
     // make a copy of class bytes so GC will not affect us.
-    char *buf = resource_allocate_bytes(THREAD, len, AllocFailStrategy::RETURN_NULL);
-    if (buf == nullptr) {
-      log_info(cds)("Out of memory when reloading class %s, quit", class_name);
-      return;
-    }
+    char *buf = NEW_RESOURCE_ARRAY(char, len);
     memcpy(buf, (char*)h_bytes->byte_at_addr(0), len);
     ClassFileStream st((u1*)buf, len, NULL, ClassFileStream::verify);
     reload_class(class_name, st, THREAD);
-
-    if (HAS_PENDING_EXCEPTION) {
-      log_info(cds)("Exception happened: %s", PENDING_EXCEPTION->klass()->name()->as_C_string());
-      log_info(cds)("Could not create InstanceKlass for class %s", class_name);
-      CLEAR_PENDING_EXCEPTION;
-      return;
-    }
+    HANDLE_IF_HAS_EXCEPTION;
```

Comment:
Is this necessary? I think bad classlists will cause errors in the JavaCalls::call_static call above, but not here.

Reply:
Here is the original code, and it is necessary, KlassFactory::create_from_stream could throw exception.

Reply:
What kid of exceptions will be thrown (other than OOM?) Do you mean `CDS.generateLambdaFormHolderClasses()` can generate bad class files?

Reply:
This is after CDS.generateLambdaFormHolderClasses, we get the byte stream of a class --- we need create instanceklass of the class. The exception is from ClassParser ---- whatever the exception during the process.

Reply:
The ClassParser will throw an error only if OOM, or if the class is invalid. However, if the class is invalid, it means there's a bug inside generateLambdaFormHolderClasses() -- it's not supposed to generate an invalid class.

So we cannot just print a warning and ignore the error. The user may not notice the error message. 

The archive dumping should fail so the user knows what has happened, and will hopefully report the error to us.